### PR TITLE
Mounted flashlight changes

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Security flashlight no longer considered a gun mod, but instead have special interactions with the gun.
Attached by clicking on a gun, removed via verb.
When attached to a gun, flashlight could be toggled via verb or new "L" hotkey, which also works for a hand-held flashlight.

Power cell in a mounted flashlight could be replaced by clicking a gun with a small power cell.
Energy guns prioritize replacing their own power cell.
So flashlight mounted on Martin pistol can't get cell replacement that way, because Martin also uses small power cells.

Also got rid of some proc redefinitions, because that's pointless extra proc calls and makes figuring out what calls what very confusing.

## Why It's Good For The Game

Having flashlight attachements is great, but implementation was lacking and pissed me off consistently during gameplay.
This PR aims to improve the situation.

## Testing

https://github.com/discordia-space/CEV-Eris/assets/65828539/78b91ba2-248e-4eb8-90cf-4f72ac962d0d

## Changelog
:cl:
add: Added "L" hotkey for toggling flashlights, standalone and mounted on a gun.
add: Added a way to quickly replace power cell in a mounted flashlight. Have to click on a gun with a proper power cell.
tweak: Changed the way flashlight attaches to guns, could be easily added and removed now.
code: Merged several gun-related proc redefinitions.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -151,6 +151,24 @@
 						amount--
 				qdel(src)
 
+	if(flashlight_attachment)
+		if(istype(I, flashlight_attachment.suitable_cell)) // TODO: Make sure no energy gun is prioritizing flashlight over it's own power cell
```
Comment (SirRichardFrancis) [inline code/modules/projectiles/gun.dm:155]:
```suggestion
		if(istype(I, flashlight_attachment.suitable_cell))
```
Have been done.

File: interface/skin.dmf [hunk 2]
```diff
@@ -681,6 +681,9 @@ macro "hotkeymode"
 	elem
 		name = "CTRL+Z"
 		command = "Activate-Held-Object"
+	elem
+		name = "L"
+		command = "Toogle-Flashlight"
```
Comment (hyperioo) [inline interface/skin.dmf:686]:
```suggestion
		command = "Toggle-Flashlight"
```

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -369,6 +369,27 @@
 	if (W)
 		W.attack_self(src)
 
+
+/mob/verb/toogle_flashlight()
+	set name = "Toogle Flashlight"
```
Comment (hyperioo) [inline code/modules/mob/mob.dm:374]:
```suggestion
/mob/verb/toggle_flashlight()
	set name = "Toggle Flashlight"
```

File: code/modules/projectiles/gun.dm [hunk 4]
```diff
@@ -151,6 +151,24 @@
...
+
+	else if(istype(I, /obj/item/device/lighting/toggleable/flashlight/seclite))
+		if(!user.unEquip(I))
+			return
+		flashlight_attachment = I
+		I.forceMove(src)
+		playsound(loc, 'sound/weapons/guns/interact/pistol_magin.ogg', 75, 1)
+		to_chat(user, SPAN_NOTICE("You attach [I] to [src]."))
+		verbs += /obj/item/gun/proc/remove_flashlight
+		verbs += /obj/item/gun/proc/toogle_flashlight
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:170]:
```suggestion
		verbs += /obj/item/gun/proc/toggle_flashlight
```

File: code/modules/projectiles/gun.dm [hunk 5]
```diff
@@ -1032,42 +1056,62 @@
...
+		H.update_cursor()
+	else
+		H.remove_cursor()
+
+/obj/item/gun/try_transfer(target, mob/living/user)
+	. = ..()
+	user.remove_cursor()
+
+/obj/item/gun/proc/toogle_flashlight()
+	set name = "Toogle flashlight"
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:1099]:
```suggestion
/obj/item/gun/proc/toggle_flashlight()
	set name = "Toggle flashlight"
```

File: code/modules/projectiles/gun.dm [hunk 6]
```diff
@@ -1032,42 +1056,62 @@
...
+	if(flashlight_attachment && ishuman(usr))
+		flashlight_attachment.attack_self(usr)
+
+/obj/item/gun/proc/remove_flashlight()
+	set name = "Remove flashlight"
+	set category = "Object"
+
+	if(flashlight_attachment && ishuman(usr))
+		verbs -= /obj/item/gun/proc/remove_flashlight
+		verbs -= /obj/item/gun/proc/toogle_flashlight
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:1111]:
```suggestion
		verbs -= /obj/item/gun/proc/toggle_flashlight
```

File: code/modules/projectiles/gun.dm [hunk 7]
```diff
@@ -1032,42 +1056,62 @@
...
+		var/mob/living/carbon/human/user = usr
+		flashlight_attachment.forceMove(get_turf(user))
+		playsound(loc, 'sound/weapons/guns/interact/pistol_magout.ogg', 75, 1)
+		to_chat(user, SPAN_NOTICE("You dettach [flashlight_attachment] from [src]."))
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:1115]:
```suggestion
		to_chat(user, SPAN_NOTICE("You detach \the [flashlight_attachment] from \the [src]."))
```

File: code/modules/projectiles/gun.dm [hunk 8]
```diff
@@ -151,6 +151,24 @@
...
+			else if(flashlight_attachment.insert_item(I, user)) // Add a missing power cell
+				flashlight_attachment.cell = I
+
+	else if(istype(I, /obj/item/device/lighting/toggleable/flashlight/seclite))
+		if(!user.unEquip(I))
+			return
+		flashlight_attachment = I
+		I.forceMove(src)
+		playsound(loc, 'sound/weapons/guns/interact/pistol_magin.ogg', 75, 1)
+		to_chat(user, SPAN_NOTICE("You attach [I] to [src]."))
```
Comment (hyperioo) [inline code/modules/projectiles/gun.dm:168]:
```suggestion
		to_chat(user, SPAN_NOTICE("You attach \the [I] to \the [src]."))
```

**Review by hyperioo — APPROVED** [review]

**Pull request merged**